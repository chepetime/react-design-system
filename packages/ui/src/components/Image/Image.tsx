import React from "react";
import classnames from "classnames";
import css from "./Image.module.scss";

export interface ImageProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Image = React.forwardRef<HTMLDivElement, ImageProps>(
  (props: ImageProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Image)} ref={ref}>
        <p>Image component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Image.displayName = "Image";
