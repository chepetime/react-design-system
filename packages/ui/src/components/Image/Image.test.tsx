
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Image } from "./Image";
import { render, screen } from "@testing-library/react";

describe("Image", () => {
  it("renders without errors", () => {
    render(<Image />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Image>Some children</Image>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Image>Some children</Image>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Image />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
