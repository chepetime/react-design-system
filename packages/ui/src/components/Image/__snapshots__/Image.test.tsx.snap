// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches snapshot 1`] = `
<div>
  <div
    class="Image"
  >
    <p>
      Image component working!
    </p>
    <div />
  </div>
</div>
`;
